packages/datadog-plugin-aws-sdk/src/services/bedrockruntime/utils.js

```diff
@@ -234,17 +236,45 @@ function extractRequestParams (params, provider) {
       })
     }
     case PROVIDER.AMAZON: {
+      const prompt = requestBody.inputText
       if (modelId.includes('embed')) {
-        return new RequestParams({ prompt: requestBody.inputText })
+        return new RequestParams({ prompt })
+      } else if (prompt !== undefined) {
+        const textGenerationConfig = requestBody.textGenerationConfig || {}
+        return new RequestParams({
+          prompt,
+          temperature: textGenerationConfig.temperature,
+          topP: textGenerationConfig.topP,
+          maxTokens: textGenerationConfig.maxTokenCount,
+          stopSequences: textGenerationConfig.stopSequences
+        })
+      } else if (Array.isArray(requestBody.messages)) {
+        const inferenceConfig = requestBody.inferenceConfig || {}
+        const messages = []
+        if (requestBody.system && Array.isArray(requestBody.system)) {
+          for (const sysMsg of requestBody.system) {
+            messages.push({
+              content: sysMsg.text,
+              role: 'system'
+            })
+          }
+        }
+        messages.push(
+          ...requestBody.messages.map(message => ({
+            content: message.content?.[0]?.text,
```

Comment:
should we filter the `requestBody.messages` to only text messages (not sure if there can be other types, like file, image, etc. that we would want to ignore)